src/language_server_protocol.rs

```diff
@@ -2045,6 +2046,27 @@ impl LanguageClient {
         Ok(())
     }
 
+    pub fn workspace_configuration(&self, params: &Value) -> Result<Value> {
+        let config_params = ConfigurationParams::deserialize(params)?;
+
+        let filename = self.vim()?.get_filename(&Value::Null)?;
+        let language_id = self.vim()?.get_language_id(&filename, &Value::Null)?;
+        let root = self.get(|state| state.roots.get(&language_id).cloned().unwrap_or_default())?;
+
+        let settings = self.get_workspace_settings(&root)?;
+
+        let configuration_items = config_params
+            .items
+            .into_iter()
+            .filter_map(|item| {
+                let section = format!("/{}", item.section?.replace(".", "/"));
+                settings.pointer(&section).cloned()
+            })
+            .collect::<Value>();
+
+        Ok(serde_json::to_value(&configuration_items)?)
```

Comment:
`configuration_items` is already a `Value`, so there's no need to `serde_json::to_value` here.